app/models/assignment.rb

```diff
@@ -236,6 +236,8 @@ def update_teamsets
         self.errors.add(:base, "The specified teamset to be copied does not exist")
         return false
       end
+      # TODO: Should we throw a separate exception here or use
+      # this one?
```

Comment:
The call to `ts.dup` enters a transaction which itself might throw an `ActiveTeamWithDroppedStudent` exception.  We need to handle that exception here by adding its message to `self.errors.add(:base, ...)`, and returning false, instead of continuing with this code.